plugin/src/storytelling.ts

```diff
@@ -0,0 +1,108 @@
+import {
+  PostMessageProps,
+  Viewport,
+  PluginMessage,
+  PluginExtensionInstance,
+  ShareStory,
+  SaveStoryData,
+  CancelPlayStory,
+} from "@web/extensions/storytelling/core/types";
+
+import html from "../dist/web/storytelling/core/index.html?raw";
+import storyeditorHtml from "../dist/web/storytelling/modals/sceneeditor/index.html?raw";
+
+const reearth = (globalThis as any).reearth;
+
+reearth.ui.show(html, { width: 89, height: 40, extended: false });
+
+let sidebarId: string;
+const getSidebarId = () => {
+  if (sidebarId) return;
+  sidebarId = reearth.plugins.instances.find(
+    (instance: PluginExtensionInstance) => instance.extensionId === "sidebar",
+  )?.id;
+};
+getSidebarId();
+
+reearth.on("pluginmessage", (pluginMessage: PluginMessage) => {
+  reearth.ui.postMessage(pluginMessage.data);
+});
+
+reearth.on("message", ({ type, payload }: PostMessageProps) => {
+  switch (type) {
```

Comment:
Sorry, unless you think it is better to do a switch statement in the on("message") can you do if, if/else like the other extensions just to conform to style?

Reply:
Okey!